### PR TITLE
Rewrite volume control

### PR DESCRIPTION
**Pull request by mharis001**
**When merged this pull request will:**
- Rewrite the volume control GUI
    - Uses a progress bar instead of a slider
    - New color scale: yellow -> orange -> red
    - Can be repositioned now
- Some general cleanup of the `sys_gui` component
    - Remove unused include files `MyDialogDefines.hpp` and `script_strings.hpp`

<details>
<summary>Images</summary>

screen_2

volume_control_comparison

</details>

### CODE REVIEW

**Comment by mharis001** [comment]
Not tested in MP and I think I changed all of the volume control things in other components.

~~Can also increase the number of steps further by changing the `VOLUME_LEVEL_CHANGE` macro.~~

**Comment by Soldia1138** [comment]
Nice addition :)
Regarding the 10 steps: We already had that discussion once or twice (and even had 9 steps at one point). The outcome was that more than 5 steps are too confusing for the majority of players and it's too hard to hear the differences between two adjacent steps. So I would prefer to stay at 5 steps unless we can provide a better and more natural feedback on how far the direct speech is being propagated (it's not that easy IRL btw.)

**Comment by mharis001** [comment]
Changed back to 5 steps. Removed white from the color scale.

**Comment by TheMagnetar** [comment]
Outstanding!

**Comment by jonpas** [comment]
Great!

File: addons/sys_gui/CfgUIGrids.hpp [hunk 1]
```diff
@@ -16,13 +26,19 @@ class CfgUIGrids {
...
             };
         };
-
         class Variables {
+            class GVAR(volumeControl) {
+                displayName = CSTRING(VolumeControlGrid_DisplayName);
+                description = CSTRING(VolumeControlGrid_Description);
+                preview = QPATHTOF(data\ui\volume_control_preview.paa);
+                saveToProfile[] = {0, 1};
+                canResize = 0;
```
Comment (jonpas) [inline addons/sys_gui/CfgUIGrids.hpp]:
```suggestion
                canResize = 1;
```
Not sure if anything else has to be changed.

It is by default much smaller now, is that by design?

**Review by jonpas — CHANGES_REQUESTED** [review]

File: addons/sys_gui/CfgUIGrids.hpp (hunk 1 above)
Comment (mharis001) [reply addons/sys_gui/CfgUIGrids.hpp]: Done. Can increase the default size if you think it should be bigger.

Comment (jonpas) [reply addons/sys_gui/CfgUIGrids.hpp]: I am ok with it being smaller, it's very visible with the rest of the changes. @IDI-Systems/acre-developers what do the rest of you think?

File: addons/sys_gui/RscTitles.hpp [hunk 2]
```diff
@@ -0,0 +1,67 @@
+class RscTitles {
+    #define VOLUME_CONTROL_POSITION \
+        x = profileNamespace getVariable ['TRIPLES(IGUI,GVAR(volumeControl),X)', VOLUME_CONTROL_DEFAULT_X]; \
```
Comment (jonpas) [inline addons/sys_gui/RscTitles.hpp]: Those should be quoted.

File: addons/sys_gui/RscTitles.hpp [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+        y = profileNamespace getVariable ['TRIPLES(IGUI,GVAR(volumeControl),Y)', VOLUME_CONTROL_DEFAULT_Y]; \
+        w = profileNamespace getVariable ['TRIPLES(IGUI,GVAR(volumeControl),W)', VOLUME_CONTROL_DEFAULT_W]; \
+        h = profileNamespace getVariable ['TRIPLES(IGUI,GVAR(volumeControl),H)', VOLUME_CONTROL_DEFAULT_H] \
```
Comment (jonpas) [inline addons/sys_gui/RscTitles.hpp]: Semi-colon here and not below.

**Review by jonpas — CHANGES_REQUESTED** [review]
A few minor things left.

**Review by jonpas — APPROVED** [review]
Looks good to me and works nicely. Requesting review from others as well since it's a big change.

**Review by Sniperhid — APPROVED** [review]
LGTM

**Review by TheMagnetar — APPROVED** [review]
Outstanding!

**Review by NouberNou — APPROVED** [review]

**Pull request merged**